src/ccnl-pkt/src/ccnl-pkt-ccntlv.c

```diff
@@ -84,14 +84,23 @@ ccnl_ccntlv_getHdrLen(unsigned char *data, int len)
     return -1;
 }
 
-// parse TL (returned in typ and vallen) and adjust buf and len
+
+/**
+ * parse TL (returned in typ and vallen) and adjust buf and len
+ * @param buf allocated buffer in which the tlv should be opened
+ * @param len length of the buffer
+ * @param typ return value via pointer: type value of the tlv
+ * @param vallen return value via pointer: length value of the tlv
+ * @return 0 on success, -1 on failure.
+ */
 int
 ccnl_ccntlv_dehead(unsigned char **buf, int *len,
                    unsigned int *typ, unsigned int *vallen)
 {
 // Avoiding casting pointers to uint16t -- issue with the RFduino compiler?
 // Workaround:
     uint16_t tmp;
+    size_t maxlen = *len;
```

Comment:
Do we need a check on the len? What happens if the ``len`` is negative. Does this affect the rest of the code in the function?

Reply:
this should be an unsigned value anyway?

Reply:
This is a mess. It is clearly out of scope of this PR, but we should tackle this signed int craziness in length parameters rather soon.